ext/gd/gd.c

```diff
@@ -3645,6 +3645,16 @@ static void php_image_filter_scatter(INTERNAL_FUNCTION_PARAMETERS)
 		Z_PARAM_ARRAY(hash_colors)
 	ZEND_PARSE_PARAMETERS_END();
 
+	if (ZEND_SIZE_T_INT_OVFL(scatter_sub)) {
+		zend_argument_value_error(3, "must not be greater than %d", INT_MAX);
+		RETURN_THROWS();
+	}
+
+	if (ZEND_SIZE_T_INT_OVFL(scatter_plus)) {
+		zend_argument_value_error(4, "must not be greater than %d", INT_MAX);
+		RETURN_THROWS();
+	}
```

Comment:
Does this also handle negative underflows? Or do the scatter arguments need to be greater than 0?